lib/Sema/CSBindings.cpp

```diff
@@ -1620,19 +1620,29 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
   PO.PrintTypesForDebugging = true;
 
   out.indent(indent);
+  std::vector<StringRef> attributes;
   if (isDirectHole())
-    out << "hole ";
+    attributes.push_back("hole");
   if (isPotentiallyIncomplete())
-    out << "potentially_incomplete ";
+    attributes.push_back("potentially_incomplete");
   if (isDelayed())
-    out << "delayed ";
+    attributes.push_back("delayed");
   if (isSubtypeOfExistentialType())
-    out << "subtype_of_existential ";
+    attributes.push_back("subtype_of_existential");
   auto literalKind = getLiteralKind();
-  if (literalKind != LiteralBindingKind::None)
-    out << "literal=" << static_cast<int>(literalKind) << " ";
+  if (literalKind != LiteralBindingKind::None) {
+    SmallString<64> literalAttrStr;
+    auto literalAttr = ("[literal: " + getLiteralBindingKind(literalKind)
+                        + "]").toStringRef(literalAttrStr);
```

Comment:
I think this is incorrect because `SmallString` (that retains allocation used for StringRef) would go out of scope after `if` is done, which means that only reason why this code works is because nothing is re-using the stack space while function is executing. You should just use `std::string` here as a mentioned above.